### PR TITLE
Update the Go Hello-World Example

### PR DESCRIPTION
**Pull request by mikepianka**
*Issue #, if available:*
https://github.com/aws/aws-sam-cli-app-templates/issues/421

*Description of changes:*
Update the Go hello-world example so that it does not rely on an external API call. The call to `https://checkip.amazonaws.com` would intermittently fail and result in unexpected errors. These changes simplify the example and get the requestor's IP from the request context.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go1.x/hello-img/{{cookiecutter.project_name}}/hello-world/main.go [hunk 1]
```diff
@@ -1,47 +1,24 @@
...
-	if resp.StatusCode != 200 {
-		return events.APIGatewayProxyResponse{}, ErrNon200Response
-	}
-
-	ip, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return events.APIGatewayProxyResponse{}, err
-	}
+	var greeting string
+	sourceIP := request.RequestContext.Identity.SourceIP
```
Comment (sriram-mv) [inline go1.x/hello-img/{{cookiecutter.project_name}}/hello-world/main.go:12]: what kind of intermittent errors are you running into? Most of our other examples also use a similar pattern. Making a change in one should also mean we should standardize across all.

Comment (sriram-mv) [reply go1.x/hello-img/{{cookiecutter.project_name}}/hello-world/main.go:12]: Read https://github.com/aws/aws-sam-cli-app-templates/issues/421 for additional context. Though I agree in principle, sounds like this should break other init templates too. 🤔

Comment (mikepianka) [reply go1.x/hello-img/{{cookiecutter.project_name}}/hello-world/main.go:12]: I typically just use the Go template so I can't speak to the behavior in others. Though for a minimal example, I do agree with @moelasmar's comment in #421 that the hello-world example is doing too much and particularly should not rely on an API call to an external service for local testing.

**Comment by mndeveci** [comment]
LGTM! Would you also interested in updating the GoLang AL2 template here? 
https://github.com/aws/aws-sam-cli-app-templates/blob/master/al2/go/hello/%7B%7Bcookiecutter.project_name%7D%7D/hello-world/main.go

**Comment by mikepianka** [comment]
> LGTM! Would you also interested in updating the GoLang AL2 template here? https://github.com/aws/aws-sam-cli-app-templates/blob/master/al2/go/hello/%7B%7Bcookiecutter.project_name%7D%7D/hello-world/main.go

Sure I can do that. On first glance it looks like all the files would be the same? I don't use AL2 typically to test that.

**Comment by mndeveci** [comment]
> > LGTM! Would you also interested in updating the GoLang AL2 template here? https://github.com/aws/aws-sam-cli-app-templates/blob/master/al2/go/hello/%7B%7Bcookiecutter.project_name%7D%7D/hello-world/main.go
> 
> Sure I can do that. On first glance it looks like all the files would be the same? I don't use AL2 typically to test that.

They should have the same GoLang code, only difference between them is in the template definition (one uses golang as runtime where other one uses provided.al2 with buildmethod option in metadata)

**Comment by mikepianka** [comment]
> > > LGTM! Would you also interested in updating the GoLang AL2 template here? https://github.com/aws/aws-sam-cli-app-templates/blob/master/al2/go/hello/%7B%7Bcookiecutter.project_name%7D%7D/hello-world/main.go
> > 
> > 
> > Sure I can do that. On first glance it looks like all the files would be the same? I don't use AL2 typically to test that.
> 
> They should have the same GoLang code, only difference between them is in the template definition (one uses golang as runtime where other one uses provided.al2 with buildmethod option in metadata)

OK the al2 Go examples are updated now as well.

**Review by mndeveci — APPROVED** [review]

**Review by moelasmar — APPROVED** [review]

**Pull request merged**